great_expectations/cli/datasource.py

```diff
@@ -125,6 +125,12 @@ def datasource_list(directory):
 
 @datasource.command(name="profile")
 @click.argument('datasource_name', default=None, required=False)
+@click.option(
+    "--generator_name",
+    "-g",
+    default=None,
+    help="The name of the batch kwarg generator configured in the datasource. The generator will list data assets in the datasource"
+)
```

Comment:
While this makes sense, it is a new feature that definitely introduces a new dimension of combinations that appear yet unproven with tests.

Reply:
You are right! Adding tests in a separate PR.